### PR TITLE
Improved DE type translations and added missing search categories

### LINKED ISSUE #3260: Category search can't find by full name
**Steps to reproduce:**

* OrgancMaps for iPad v.2022.08.23-3, or OM for Android v.2022.07.27-3
* Set Ukrainian locale
* Open Organic maps and create new POI
* In categories search enter "Витвір"

**Expected result:** Category "Витвір мистецства" (which means Artwork) is found
**Actual result:** No categories found

But if I search by "Твір мистецтва" or "Скульптура" then category is found:

Looks like `NewFeatureCategories::Search(std::string const & query)` method only looks for string from `categories.txt` but ignores `type_strings.txt`.

### LINKED ISSUE #8290: Search by displayed category / feature types names doesn't always work
When adding a new POI, searching for a category doesn't correctly match category names. For instance, searching "Home improvement store" comes up empty, but when searching "hardware", "Home improvement store" is displayed.

At a guess, it seems to only match synonyms or tags of the categories, but not the names themselves?

Android 13
2024.05.11-3-FDroid

Doesn't match name            |  Matches synonym
:-------------------------:|:-------------------------:
Screenshot_20240525-154801 | Screenshot_20240525-154806 
Screenshot_20240525-154746 | Screenshot_20240525-154739

### PR DESCRIPTION
**Pull request by map-per**
- Improved German type translations
- added missing search categories to fix  #8290 for the German language (translations for new categories are copied form _types_strings.txt_)

The tests fail due to arbitrary maximum bounds in the test cases. Should I just increase the bounds?

<details><summary></summary>
<p>

Todo: @category_tourism, Lebensmittelkonserven, historic-mine, test cases, remove categories

</p>
</details>

### CODE REVIEW

File: data/strings/types_strings.txt [hunk 1]
```diff
@@ -25947,7 +25948,7 @@
     bg = Хранителни стоки
     cs = Potraviny
     da = Købmand
-    de = Lebensmittelgeschäft
+    de = Unverderbliche Lebensmittel Geschäft
```
Comment (Misalf-git) [inline data/strings/types_strings.txt:25951]: "Unverderbliche Lebensmittel Geschäft" doesn't seem correct.
I think "Lebensmittelgeschäft" is fine.

**Comment by Misalf-git** [comment]
LGTM, except "Unverderbliche Lebensmittel Geschäft".

File: data/strings/types_strings.txt (hunk 1 above)
Comment (map-per) [reply data/strings/types_strings.txt:25951]: "Lebensmittelgeschäft" doesn't match the OSM definition of [shop=grocery](https://wiki.openstreetmap.org/wiki/Tag:shop%3Dgrocery). But "Haltbare Lebensmittel" is a more concise name.

(Using the imprecise "Lebensmittelgeschäft" is a problem when adding places in the editor, but maybe we should just remove this niche category there (can_add=no) anyway.)

**Comment by pastk** [comment]
> The tests fail due to arbitrary maximum bounds in the test cases. Should I just increase the bounds?

@vng could you please take a look?

File: data/categories.txt [hunk 2]
```diff
@@ -4137,6 +4258,46 @@ vi:Phế tích
 zh-Hans:废墟|景点
 zh-Hant:1廢墟|古蹟|遺址|旅遊景點
 
+istoric-mine
```
Comment (pastk) [inline data/categories.txt:4261]:
```suggestion
historic-mine
```

**Review by pastk — APPROVED** [review]

**Review by vng — COMMENTED** [review]
- Do we really need to search mast, flagpole?
- Add ```@category_tourism``` to newly added _major_ historic-* types? At least for locomotive, I think.

**Comment by vng** [comment]
Discuss: Is it better to unite all historic-* types into one search category? Or keep only the most _interesting_ separate categories? I doubt that someone will specifically search for the stone or wreck or mine ..

The reason here is to keep the category search index with some reasonable size.

**Comment by map-per** [comment]
> Do we really need to search for mast or flagpole?

Yes, to make the search in the editor category list work (see:  #3260). It is definitely a bug that you don't find e.g. _wreck_ in the list when you search for "wreck".

> Add @category_tourism to newly added major historic-* types? At least for locomotive, I think.

Good idea, I think it definitely makes sense for locomotive, tank, and aircraft and as artworks and sculptures are in @category_tourism we should also add stone, cannon, and anchor. So only the dangerous places, mine and wreck, are left out then.

> The reason here is to keep the category search index with some reasonable size.

I don't know enough about the search algorithm to judge that. But we already have around 450 categories, so adding 26 more doesn't feel like a big deal to me. And it would be more consistent to add search terms for all categories used in the editor and not just for most.

**Comment by vng** [comment]
- Well, Editor should not depend on the search category list. It should be refactored definitely, and AFAIR I made an issue for that. So you should able to add/edit wrecks and flagpoles, but no need to search for them.
- Let's keep the search categories list reasonable.

**Comment by map-per** [comment]
Developing a separate search just for the editor sounds like an unnecessary complicated solution to me. We only have a hand full of categories that might not be interesting for map search (e.g. pole). And even there, offering a search for pole, just like for most other categories, is a more consistent user experience. 26 categories more is not an unreasonable number IMO. Do you know the runtime complexity of the algorithm? Is it $O(log(n))$?

What's interesting is, that the _AddAllCategoriesInAllLangs()_ that causes the unit tests to fail because of a high number of nodes is not used in the app. Search only uses the users language and en.

**Comment by vng** [comment]
- This is not about complexity, but about possible confusing/irrelevant matches, due to separate tokens processing when searching.
- I will take a look, but there is no need to write a separate search for the Editor. The Editor should use type_strings instead of categories. Categories file is _not intended_ to be used in the Editor, it was made like "lets put something fast here" ..

File: data/strings/types_strings.txt (hunk 1 above)
Comment (map-per) [reply data/strings/types_strings.txt:25951]: I still prefer "Haltbare Lebensmittel" because not all long lasting good are packaged in cans.

**Comment by pastk** [comment]
> Good point. Yes, we can also merge categories into Editor, but not vice-versa e.g. filling categories for the Editor purposes.

How do you propose to do that?
Maintain a separate synonyms list just for the Editor? (for types which are not supposed to be searched on the map)

We need synonyms in the Editor search as the list is too long to be browse-able.
(e.g. I had a hard time to find a `historic-wreck` type in the Editor list: there is no "Wreck" in the list and searching by "wreck" brings nothing - ultimately I had to lookup types_strings.txt where it was translated as "Shipwreck"..)

**Comment by pastk** [comment]
Linking related issues (in addition to #3260 ):
- https://github.com/organicmaps/organicmaps/issues/8290
- https://github.com/organicmaps/organicmaps/issues/3496

File: data/strings/types_strings.txt (hunk 1 above)
Comment (Misalf-git) [reply data/strings/types_strings.txt:25951]: Wegen _Konservendose_? Konservieren geht auch mit Einmachgläsern oder pökeln, trocknen etc.

Comment (map-per) [reply data/strings/types_strings.txt:25951]: Oh, stimmt. Dann können wir "Lebensmittelkonserven" nehmen.

**Comment by map-per** [comment]
@vng I'm still not convinced that adding about 5 to 9 categories that are not relevant for map search is a problem. But search is your field. Which categories should I remove from this PR?

**Comment by vng** [comment]
- Remove flagpole, must.
- Add ```@category_tourism``` to the locomotive, airplane, tank.
- Ok, keep your newly added historic types as-is

Thank you.

**Comment by vng** [comment]
Seems like locomotive, airplane, tank should be added to the ```AttractionsChecker::AttractionsChecker```

**Review by vng — APPROVED** [review]

**Comment by map-per** [comment]
Great, thank you for the review @pastk, @vng, and @Misalf-git. 

I will check the English search categories next.

**Comment by biodranik** [comment]
> But its useful for OSMers/mappers and devs/testers

These guys can always use any other software to get what's needed. It's better to keep OM focused on simplicity and efficiency for non-geeky users.

**Comment by map-per** [comment]
Don't worry, all added search categories are also interesting for general map users

**Comment by pastk** [comment]
> These guys can always use any other software to get what's needed. It's better to keep OM focused on simplicity and efficiency for non-geeky users.

How a user would add a Flagpole or a Mast via our Editor? They're not showing up via search there, thats the issue.

**Pull request merged**